### DISCUSSION TITLE
error 413: Entity too Large

### DISCUSSION THREAD

**Discussion by dim9-git** [vercel/next.js#56920] [App Router]
I am getting error while requesting inner API method with JSON payload from a form that exceeds 1mb

I tried adding to my module.exports in next.config.js a new propety called

api: {
    bodyParser: {
      sizeLimit: "10mb",
    },
  },

but it seems not to work with next.js 13, since I am getting warning in my terminal:

 ⚠     The root value has an unexpected property, api, which is not in the list of allowed properties (amp, analyticsId, assetPrefix, basePath, cleanDistDir, compiler, compress, configOrigin, crossOrigin, devIndicators, distDir, env, eslint, excludeDefaultMomentLocales, experimental, exportPathMap, generateBuildId, generateEtags, headers, httpAgentOptions, i18n, images, modularizeImports, onDemandEntries, optimizeFonts, output, outputFileTracing, pageExtensions, poweredByHeader, productionBrowserSourceMaps, publicRuntimeConfig, reactProductionProfiling, reactStrictMode, redirects, rewrites, sassOptions, serverRuntimeConfig, skipMiddlewareUrlNormalize, skipTrailingSlashRedirect, staticPageGenerationTimeout, swcMinify, target, trailingSlash, transpilePackages, typescript, useFileSystemPublicRoutes, webpack).

**Comment by icyJoseph** [comment]
Hi,

AFAIK, the bodyParser size limit does not work on App Router.

Also, you don't set that in `next.config.js`, but on the Pages Router API route files, so for example you'd have:

```ts
import type { NextApiRequest, NextApiResponse } from 'next'
 
export default function handler(req: NextApiRequest, res: NextApiResponse) {
  if (req.method === 'POST') {
    // Process a POST request
  } else {
    // Handle any other HTTP method
  }
}

export const config = {
  api: {
    bodyParser: {
      sizeLimit: '1mb',
    },
  }
}
```

Something like that. So it is per route configuration. That's why you are getting that warning about badly shaped next.config.js ~

Again, to be clear, as far as I know, as of today, there's no way to control that in Router Handlers (App Router) ~ I haven't been able to find it in the docs.

**Comment by Karthickeyyan** [comment]
Hi All,

I might have a simple solution for this. Let's say the Next.js application is deployed on either Azure Static Web Apps or Amplify; both are serverless architectures.

I’ve also tried all these options, but none of them worked for me. You can check out this blog to reach a conclusion.

https://medium.com/@karthickeyyan.perumalsamy/effective-strategies-for-handling-large-payloads-in-next-js-48206f0450bf
